ethereum/check_swap_factory_contract_test.go

```diff
@@ -93,18 +102,36 @@ func TestCheckSwapFactoryContractCode(t *testing.T) {
 
 // Tests that we fail when the wrong contract byte code is found
 func TestCheckSwapFactoryContractCode_fail(t *testing.T) {
-	ec, chainID := tests.NewEthClient(t)
+	ec, _ := tests.NewEthClient(t)
 	pk := tests.GetMakerTestKey(t)
-	txOpts, err := bind.NewKeyedTransactorWithChainID(pk, chainID)
-	require.NoError(t, err)
 
-	// Deploying an arbitrary contract that won't match the swap factory contract
-	contractAddr, tx, _, err :=
-		DeployERC20Mock(txOpts, ec, "ERC20Mock", "MOCK", ethcommon.Address{0x1}, big.NewInt(9999))
-	require.NoError(t, err)
-	_, err = block.WaitForReceipt(context.Background(), ec, tx.Hash())
+	// Deploy a forwarder contract and then try to verify it as SwapFactory contract
+	contractAddr := deployForwarder(t, ec, pk)
+	_, err := CheckSwapFactoryContractCode(context.Background(), ec, contractAddr)
+	require.ErrorIs(t, err, errInvalidSwapContract)
+}
+
+func TestGoerliContract(t *testing.T) {
```

Comment:
I was trying to make this test file a "one stop shop" to help any of us make all needed updates after the contracts change.  For now you have to manually enable the test unless you want me to enable it by default.

Reply:
this isn't really a test but just redeploys the contract if its code is out of date, right? maybe it should have a comment that explains that?

Reply:
Unless you add a goerli private key, it only checks the address.  The reason for the skip, is because it is connecting to an outside gonerli endpoint that we don't know much about. It seems reliable so far. Did you have an opinion on removing the `t.Skip` from the test or keeping it?  Another option might be to use an environment variable for an Infura API key that we test for in the golang code.  It looks like there is secrets store for github actions: https://docs.github.com/en/rest/actions/secrets